File: mario_less.c
#include <stdio.h>

int main(void)
{
    int height;

    do
    {
        printf("Height: ");
        scanf("%d", &height);
    }
    while (height < 1 || height > 8);

    for (int row = 1; row <= height; row++)
    {
        for (int col = 0; col < height - row; col++)
        {
            printf(" ");
        }
        for (int col = 0; col < row; col++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario_less.py
def get_height():
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                return height
        except ValueError:
            pass


def main():
    height = get_height()
    for row in range(1, height + 1):
        print(" " * (height - row) + "#" * row)


if __name__ == "__main__":
    main()
